specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,282 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+@useAuth(
+  OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit,
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/authorize",
+
+      // add description to below: "impersonate your user account"
+      scopes: ["https://eventhubs.azure.net/.default"],
+    }
+  ]>
+)
+@service({
+  title: "Azure Schema Registry",
+})
+
+// Supported operations.
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/$schemas/{id}?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/{version}?api-version={apiVersion}
+//  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/:get-id?api-version={apiVersion}
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+@server(
+  "{fullyQualifiedNamespace}",
+  "The Schema Registry service endpoint.",
+  {
+    @doc("The Schema Registry service endpoint, for example 'https://my-namespace.servicebus.windows.net'.")
+    fullyQualifiedNamespace: url,
+  }
+)
+@versioned(ServiceApiVersions)
+@doc("SchemaRegistryClient is a client for registering and retrieving schemas from the Azure Schema Registry service.")
+namespace SchemaRegistry;
+
+@doc("Represents the Schema Registry API version to use for requests.")
+enum ServiceApiVersions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2021-10 Version")
+  V2021_10: "2021-10",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2022-10 Version")
+  V2022_10: "2022-10",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2023-07-01 Version. This is the default version.")
+  V2023_07_01: "2023-07-01",
```

Comment:
I just realized this is a preview version, this api version looks like it's a GA one, the service shipped this without the `-preview` suffix?

Reply:
I just took a look at the existing swaggers and this is a GA version, why are we outputting it to a preview folder?

Reply:
Travis mentioned that I should do this, but I'm not sure if I made it clear that this was an already stable version. Will change to stable, if that's the right move!

Reply:
I also did this earlier b/c the emitted swagger was overwriting the existing one, so I didn't know it that was the right thing.

Reply:
Probably just wasnt clear, this should replace the existing swagger with the generated one